### ISSUE TITLE
Deprecation warnings with Dateutil 2.5

### ISSUE THREAD

**Issue by jenshnielsen** [matplotlib/matplotlib#6155]
After the release of Dateutil 2.5 I am seeing a number of Deprecation warnings when running the test suite:

It's not a big issue at the moment but we should preferably fix this before 2.0 so we remain compatible with future versions of Dateutil. 

```
/Users/jhn/Envs/mplmaster/lib/python3.5/site-packages/dateutil/rrule.py:415: DeprecationWarning: Using both 'count' and 'until' is inconsistent with RFC 2445 and has been deprecated in dateutil. Future versions will raise an error.
```

**Comment by pganssle** [comment]
[This line](https://github.com/matplotlib/matplotlib/blob/73f90a23cba264ed0a68026dc1d5504e7c0e684a/lib/matplotlib/dates.py#L833) is likely the culprit. I think probably the best solution is to incorporate the old behavior into the rrulewrapper.

**Comment by tacaswell** [comment]
Can we just drop the use of `count` there?

**Comment by jenshnielsen** [comment]
@tacaswell  I thinks so. The code clearly already takes into account the fact that the number of ticks may be larger than `MAXTICKS` and thus larger than `count`

**Comment by jenshnielsen** [comment]
fixed by #6190
